rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1299,36 +1179,65 @@ public NodeState getNodeState(String nodeUrl) {
+            String policyType, Object[] policyParams, boolean nodesRecoverable) {
         if (nodeSourceName == null) {
             throw new IllegalArgumentException("Node Source name cannot be null");
         }
+
+        List<Serializable> serializableInfraParams = getSerializableParamsOrFail(nodeSourceName, infraParams);
+        List<Serializable> serializablePolicyParams = getSerializableParamsOrFail(nodeSourceName, policyParams);
+
+        logger.info("Define a node source : " + nodeSourceName);
+
         nodeSourceName = nodeSourceName.trim();
 
-        NodeSourceData nodeSourceData = new NodeSourceData(nodeSourceName,
-                                                           infrastructureType,
-                                                           infrastructureParameters,
-                                                           policyType,
-                                                           policyParameters,
-                                                           caller,
-                                                           nodesRecoverable);
-        boolean added = dbManager.addNodeSource(nodeSourceData);
+        NodeSourceData persistedNodeSource = new NodeSourceData(nodeSourceName,
+                                                                infrastructureType,
+                                                                serializableInfraParams,
+                                                                policyType,
+                                                                serializablePolicyParams,
+                                                                caller,
+                                                                nodesRecoverable,
+                                                                NodeSourceStatus.NODES_UNDEPLOYED);
+
+        boolean added = dbManager.addNodeSource(persistedNodeSource);
 
+        if (!added) {
+            return new BooleanWrapper(false);
+        } else {
+            return validateAndPrepareNodeSource(nodeSourceName, persistedNodeSource, added);
+        }
+
+    }
+
+    private List<Serializable> getSerializableParamsOrFail(String nodeSourceName, Object[] params) {
+        List<Serializable> serializableParams = null;
+        if (params != null) {
+            Object currentParam = null;
+            try {
+                serializableParams = new ArrayList<>(params.length);
+                for (int i = 0; i < params.length; i++) {
+                    currentParam = params[i];
+                    serializableParams.add(i, (Serializable) currentParam);
+                }
+            } catch (ClassCastException e) {
+                throw new IllegalArgumentException("Parameter " + currentParam + " of defined node source " +
+                                                   nodeSourceName + " is not Serializable");
```

Comment:
you  should chain the original exception

Reply:
yes. done.